packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -177,6 +177,11 @@ implements ContextManager {
   ) {
     const contextManager = this;
     return function (this: never, event: string, listener: Func<void>) {
+      // @ts-expect-error listener is not a property of type Function but it is actually used by the onceWrapper in events.js
+      if (typeof listener === 'function' && typeof listener.listener === 'function' && listener.name === 'bound onceWrapper') {
```

Comment:
be aware that OTel context manager is likely not be the only one patching `EventEmitter`. More or less any APM product and a lot other tools (e.g. zone.js - yes it's used by node.js apps not only browsers) patch this also.

Therefore you might have others between `once` and `on`.

You could rely on the single threaded nature of JS and set some `inOnce` flag in ContextManager and check this one instead.

Reply:
> You could rely on the single threaded nature of JS and set some inOnce flag in ContextManager and check this one instead.

Good idea. Opened a third PR https://github.com/open-telemetry/opentelemetry-js/pull/3132